lib/state/appActions.js

```diff
@@ -42,6 +43,29 @@ appActions.trackedState = {
   SEE_IN_BLIP : 'Clicked See Data in Blip'
 };
 
+appActions.errorText = {
+  E_READING_FILE : 'Error reading file: ',
+  E_WRONG_FILE_EXT : 'Please choose a file ending in: ',
+  E_UPLOAD_IN_PROGRESS : 'Cannot start upload while an upload is in progress',
+  E_UNSUPPORTED_TYPE : 'Unsupported upload source type: ',
+  E_INVAILD_UPLOAD_INDEX : 'Invalid upload index: ',
+  E_DEVICE_NOT_CONNECTED : 'The device doesn\'t appear to be connected'
+};
+
+appActions.errorStage = {
+  STAGE_SETUP : { Code: 'E_SETUP' , Message: 'Error during setup', Step: '' },
+  STAGE_LOGIN : { Code: 'E_LOGIN' , Message: 'Error during login', Step: '' },
+  STAGE_LOGOUT : { Code: 'E_LOGOUT' , Message: 'Error during logout', Step: '' },
+  STAGE_DETECT_DEVICES : { Code: 'E_DETECTING_DEVICES' , Message: 'Error detecting devices', Step: '' },
```

Comment:
Device detection is a significant rathole, it turns out. Trying to identify a device safely and quickly when they don't share a protocol is a problem that goes up in complexity with the square of the number of devices you support. So yes, we can reasonably remove the code that tries to do it.

(Although the old saw "you can always go back in git history to recover it" is a lot harder than it looks. Reality is that if any significant amount of time has passed since you deleted it, it's rarely worth going back to try to reuse old code -- too much of the landscape will have changed.)